.github/workflows/unit_test.yml

```diff
@@ -2,7 +2,7 @@ name: Unit test
 
 on:
   push:
-    branches: [master]
+    branches: [main]
   pull_request:
   workflow_dispatch:
 
```

Comment:
the `lint-and-type-check` job is executed only on Python 3.7. It shouldn't cause any issue unless there is a breaking change in the linter in between python versions. Should we change it? What do you think?

Reply:
That is on purpose, to ensure backwards compatibility.

Reply:
That change could also be done in a separate PR, since it's irrelevant to this PR.